lib/dataObjects/apiResponseObjects/VodData.js

```diff
@@ -6,46 +6,43 @@ const KeyAndValue = require('../SimpleDataObjects/KeyAndValue');
 class VodData
 {
 
-	constructor(partnerId, flavorIds, entry, uiConf, cuePointListResult, flavorURLs)
+	constructor(partnerId, flavorDataList, selectedFlavorIds, entry, uiConf, cuePointList)
 	{
 		this.entry = entry;
-		this.cuePointList = cuePointListResult;
-		const flavorsData = [];
-		// we start at 2 since 0 was entry and 1 was cuePointslist
-		for (let i = 0; i < flavorIds.length; i++)
-		{
-			const flavorId = flavorIds[i];
-			const flavorPath = flavorURLs[i];
-			const flavorData = new KeyAndValue(flavorId, flavorPath);
-			flavorsData.push(flavorData);
-		}
-		this.flavorDataList = flavorsData;
-		this.numOfFlavors = flavorIds.length;
+		this.cuePointList = cuePointList;
+		this.flavorDataList = flavorDataList;
 		this.partnerId = partnerId;
 		this.uiConf = uiConf;
+		this.selectedFlavorIdList = selectedFlavorIds;
 	}
 
 	/**
 	 * returns only the flavors URLs as an array
 	 * @returns {Array}
 	 */
-	getOnlyFlavorPaths()
+	getOnlySelectedFlavorPaths()
 	{
 		const answer = [];
-		for (let i = 0; i < this.flavorDataList.length; i++)
-			answer.push(this.flavorDataList[i].value);
+		for (let i = 0; i < this.selectedFlavorIdList.length; i++)
+		{
+			for (let j = 0; j < this.flavorDataList.length; j++)
+			{
+				if (this.flavorDataList[j].id === this.selectedFlavorIdList[i])
```

Comment:
This can be done more optimized with one loop.

if (this.selectedFlavorIdList.indexOf(this.flavorDataList[j].id) > -1 )  do push...........